source/module_base/global_file.cpp

```diff
@@ -44,6 +45,7 @@ void ModuleBase::Global_File::make_dir_out(
 
     GlobalV::global_out_dir = prefix + suffix + "/";
     GlobalV::global_stru_dir = GlobalV::global_out_dir + "STRU/";
+    GlobalV::global_matrix_dir = GlobalV::global_out_dir + "matrix_HS/";
```

Comment:
I suggest not merging this pr for now. It seems the user who started the original issue #1237 intends to use it along with DeepH. That's why he requested we create a new directory for each file. As far as I know, it would be more convenient for them if we do things this way. Let's wait till we can get in touch with the user and see what to do.

```suggestion
    GlobalV::global_matrix_dir = GlobalV::global_out_dir + "matrix_HS/" + std::to_string(istep);
```